src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -650,6 +653,66 @@ private boolean matches(Offer offer, Request request) {
         }
     }
 
+    /**
+     * Retrieves the slaveAttributes corresponding to label name. In the event of a match on label names,
+     * returns attributes for SlaveInfo marked as default, otherwiese the matched label's slave attributes. Keeping with
+     * the previous behavior, will return null/empty slave attributes.
+     * @param labelName MesosSlaveInfo label string name.
+     * @return slaveAttributes as a JSONObject.
+     */
+
+    private JSONObject getSlaveAttributeForLabel(String labelName) {
+        LOGGER.info(String.format("Getting slave attributes for label: %s", labelName));
+        List<MesosSlaveInfo> matchedSlaves = getMesosCloud().getSlaveInfos().stream()
+                .filter(slaveInfo -> slaveInfo.getLabelString() != null && slaveInfo.getLabelString().equals(labelName))
+                .collect(Collectors.toList());
+
+        if (matchedSlaves.size() > 1) {
+            LOGGER.info(String.format("Multiple matches for label %s found, matches: %o", labelName, matchedSlaves.size()));
+            LOGGER.info("For multiple matches, checking for Slave Info marked as default");
+            return matchedSlaves.stream()
+                    .filter(MesosSlaveInfo::isDefaultSlave).collect(Collectors.toList())
+                    .stream()
```

Comment:
Is this required?

Reply:
I believe so, in testing I found that one could define multiple labels in config with same name. In this case, I check for one of the matches to be marked as default, otherwise in event of single match label proceeds with retrieving slave attributes of the matched label and if not present an empty JSONObject returned.